File: problems/3159_find_occurrences_of_an_element_in_an_array/3159_find_occurrences_of_an_element_in_an_array.py
class Solution:
    def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
        indices = [i for i, val in enumerate(nums) if val == x]
        ans = []
        for q in queries:
            if q <= len(indices):
                ans.append(indices[q - 1])
            else:
                ans.append(-1)
        return ans

File: problems/3159_find_occurrences_of_an_element_in_an_array/3159_find_occurrences_of_an_element_in_an_array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        List<Integer> indices = new ArrayList<>();
        int[] res = new int[queries.length];
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == x) {
                indices.add(i);
            }
        }
        for (int i = 0; i < queries.length; i++) {
            if (queries[i] > indices.size()) {
                res[i] = -1;
            } else {
                res[i] = indices.get(queries[i] - 1);
            }
        }
        return res;
    }
}